projects/aca-content/src/lib/components/knowledge-retrieval/search-ai/agents-button/agents-button.component.spec.ts

```diff
@@ -169,55 +206,87 @@ describe('AgentsButtonComponent', () => {
                   key: 'Enter'
                 });
           agents$.next(agentsWithAvatar);
+          agents$.complete();
+          spyOn(window, 'open');
+          notificationService = TestBed.inject(NotificationService);
+          spyOn(notificationService, 'showInfo');
+          message = 'Some message';
         });
 
-        it('should display notification if checkSearchAvailability from SearchAiService returns message', () => {
-          const message = 'Some message';
-          checkSearchAvailabilitySpy.and.returnValue(message);
-          const notificationService = TestBed.inject(NotificationService);
-          spyOn(notificationService, 'showInfo');
-          fixture.detectChanges();
+        describe('with selected nodes', () => {
+          beforeEach(() => {
+            selectionState.isEmpty = false;
+          });
 
-          getAgentsButton().dispatchEvent(event);
-          expect(notificationService.showInfo).toHaveBeenCalledWith(message);
-        });
+          it('should display notification if checkSearchAvailability from SearchAiService returns message', () => {
+            checkSearchAvailabilitySpy.and.returnValue(message);
+            fixture.detectChanges();
 
-        it('should not display notification if checkSearchAvailability from SearchAiService returns empty message', () => {
-          const message = '';
-          checkSearchAvailabilitySpy.and.returnValue(message);
-          const notificationService = TestBed.inject(NotificationService);
-          spyOn(notificationService, 'showInfo');
-          fixture.detectChanges();
+            getAgentsButton().dispatchEvent(event);
+            expect(notificationService.showInfo).toHaveBeenCalledWith(message);
+          });
 
-          getAgentsButton().dispatchEvent(event);
-          expect(notificationService.showInfo).not.toHaveBeenCalled();
-        });
+          testButtonActions();
 
-        it('should disable menu triggering if checkSearchAvailability from SearchAiService returns message', () => {
-          checkSearchAvailabilitySpy.and.returnValue('Some message');
-          fixture.detectChanges();
+          it('should enable menu triggering if checkSearchAvailability from SearchAiService returns empty message', () => {
+            checkSearchAvailabilitySpy.and.returnValue('');
+            fixture.detectChanges();
 
-          getAgentsButton().dispatchEvent(event);
-          fixture.detectChanges();
-          expect(getMenuTrigger()).toBeNull();
-        });
+            getAgentsButton().dispatchEvent(event);
+            fixture.detectChanges();
+            const menuTrigger = getMenuTrigger();
+            expect(menuTrigger).toBeTruthy();
+            expect(menuTrigger).toBe(getMenu());
+          });
+
+          it('should call checkSearchAvailability from SearchAiService with correct parameter', () => {
+            fixture.detectChanges();
```

Comment:
can we move  `fixture.detectChanges();` to beforeEach?

Reply:
done